src/ensemble_scheduler/ensemble_scheduler.cc

```diff
@@ -1347,6 +1442,9 @@ EnsembleScheduler::Enqueue(std::unique_ptr<InferenceRequest>& request)
   std::shared_ptr<EnsembleContext> context(new EnsembleContext(
       metric_reporter_.get(), stats_aggregator_, is_, info_.get(), request,
       stream_));
+#ifdef TRITON_ENABLE_STATS
+  info_->ensemble_start_ns = CaptureTimeNs();
```

Comment:
Is this `ensemble_end_ns - ensemble_start_ns` going to include time spent in the Queue?

Reply:
I think sending a couple requests for the different scenarios (cache miss, cache hit), gathering all the numbers, and analyzing them to make sense would be a good step for validation here. I can help you go over the numbers after you've collected some samples and compared against your expectations.

Reply:
It doesn't capture time spent in Queue.
Request start time in the queue is captured at line 1397.
Let me know if it should capture queue time too.
Also, to avoid any confusion about time, I can remove cache lookup time and cache insert time and capture everything in ensemble_start_ns and ensemble_end_ns_.

Reply:
The queue **start** time is captured there, but everything from there up until the compute input starts (queue end time) is considered time spent in the queue. 

Let's chat about this on Monday, we can go over a couple examples on a whiteboard or something.

Reply:
So I believe `DynamicBatchScheduler` today reports stats like this:

- Enqueue()
- request->CaptureQueueStart()
- CacheLookup() -> lookup_time = lookup_end - lookup_start
    - HIT: UpdateSuccessCacheHit()
        - queue_time = lookup_start - queue_start
        - cache_hit_time = lookup_time
    - MISS: UpdateSuccessCacheMiss()
        - (backend computation)
        - queue_time = compute_input_start - queue_start
        - (finish backend response)
        - insert_time = cache_insert_end - cache_insert_start
        - cache_miss_time =  lookup_time + insert_time (does not include compute time)

Summary:
- So in the case of cache miss, we do include some minor time spent doing lookup (should be very fast in cache miss case since there is no data copy/etc.) in the "queue time" since we do (queue_time = compute_input_start - queue_start, and compute_input_start will begin after cache_lookup_end)
- However, for cache_hit_time and cache_miss_time metrics, we only include the time spent doing actual cache things (lookup/insertion), and do not include other time spent. 
    - So I believe if we want the stats to be comparable for the ensemble and non-ensemble cases, we should do something similar
    - It looks like we [currently](https://github.com/triton-inference-server/core/blob/563a0f80a6d107c5a5fb78e46bb8bbe49538cc9a/src/ensemble_scheduler/ensemble_scheduler.cc#L1082) do: `uint64_t cache_miss_ns = lookup_ns + ensemble_ns + insert_ns;` but instead should maybe be `uint64_t cache_miss_ns = lookup_ns + insert_ns;`. What do you think?

Reply:
Fixed the calculation.
Numbers to be shared personally over email/slack.